app/src/main/java/com/lagradost/cloudstream3/ui/settings/SettingsAccount.kt

```diff
@@ -133,7 +142,85 @@ class SettingsAccount : PreferenceFragmentCompat(), BiometricAuthenticator.Biome
+                                        .render().nativeImage() as Bitmap
+
+                                    activity.runOnUiThread {
+                                        dialog.show()
+                                        binding.devicePinCode.setText(txt(pinCodeData.userCode))
+                                        binding.deviceAuthMessage.setText(txt(R.string.device_pin_url_message, pinCodeData.verificationUrl))
+                                        binding.deviceAuthQrcode.setImage(
+                                            img(qrCodeImage)
+                                        )
+                                    }
+
+                                    var expirationCounter = pinCodeData.expiresIn
+
+                                    while (expirationCounter > 0) {
+                                        activity.runOnUiThread {
+                                            binding.deviceAuthValidationCounter.setText(
+                                                txt(
+                                                    R.string.device_pin_counter_text, expirationCounter.div(60) , expirationCounter.rem(60)
+                                                )
+                                            )
+                                        }
+
+                                        if (expirationCounter.rem(pinCodeData.interval)  == 0 && api.handleDeviceAuth(pinCodeData)) {
+                                            activity.runOnUiThread {
+                                                    showToast(
+                                                        activity.getString(R.string.authenticated_user)
+                                                            .format(
+                                                                api.name
+                                                            )
+                                                    )
+                                                dialog.dismissSafe(activity)
+                                            }
+                                            return@ioSafe
+                                        }
+                                        delay(1000)
+                                        expirationCounter--
+                                    }
+                                    activity.runOnUiThread {
+                                        showToast(
+                                            activity.getString(R.string.device_pin_expired_message)
+                                        )
+                                        dialog.dismissSafe(activity)
+                                    }
+                                } catch (e: Exception) {
+                                    logError(e)
+                                }
+                            }
+                        } else {
+                            api.authenticate(activity)
```

Comment:
this is really weird to read, reverse this if statment so this is on top and else is the rest of the code